### DISCUSSION TITLE
Newbie-Question: How to get links to the headings of the note?

### DISCUSSION THREAD

**Discussion by merlinuwe** [chrisgurney/obsidian-note-toolbar#36] [General]
Thanks for your interesting plugin.

I have some notes (HH22AB INF, HH23A IW, ...) in a folder. They all have the same headings:

## ToDo

How can I get a toolbar in all of these files which stays visible at the top of the note when I scroll the page down and up?

I tried this:

`````
> [!note-toolbar|sticky-border]
> - [[HH22AB INF#^492b2c|Termine]]
> - [[HH22AB INF#^aktuellen|Noch folgender U.]]
`````

**Comment by chrisgurney** [comment]
You're close! 

First off, I'm not sure if the backticks are in your note or not, so make sure they're not.

Without the backticks, this format works:

```
> [!note-toolbar|sticky-border]
> - [[Filename#HeadingName1|Alias1]]
> - [[Filename#HeadingName2|Alias2]]
```

So I believe in your case it's:

```
> [!note-toolbar|sticky-border]
> - [[HH22AB INF#Termine|Termine]]
> - [[HH22AB INF#Noch folgender U.|Noch folgender U.]]
```

I'm not sure what `^492b2c` and `^aktuellen` relate to in your notes.

(Note that the `sticky` style isn't working in Reading/Preview mode at the moment, but I'm working on a fix for that.)

Hope this helps!

**Comment by merlinuwe** [comment]
Thanks for your reply. In the meantime, I found an other solution, which holds the menu (at the moment in edit mode only) at the top of the note. 

![image](https://github.com/chrisgurney/obsidian-note-toolbar/assets/4138869/23e2e583-80df-4bc7-883a-ab7b37ba078e)

I like this (I had to restart Obsidian or close and open the note to see the result):

![image](https://github.com/chrisgurney/obsidian-note-toolbar/assets/4138869/5d1b0bcc-3020-4799-9809-c0b1f6ea1872)


Here is my data.json file which configures the magic:

[data.json](https://github.com/chrisgurney/obsidian-note-toolbar/files/14951251/data.json)

---

Screenshots from different themes:

Minimal theme:

![image](https://github.com/chrisgurney/obsidian-note-toolbar/assets/4138869/3eeb0dee-7542-4eee-933d-83e367bcbdef)

magic user theme:

![image](https://github.com/chrisgurney/obsidian-note-toolbar/assets/4138869/f831da88-3f70-4b5c-8e57-bf30e7422b6b)

I'm looking forward to the sticky style. ;-)

---

When I'm at the top of the note, the menu looks like this:

![image](https://github.com/chrisgurney/obsidian-note-toolbar/assets/4138869/9d9b27da-6d56-4a00-b856-44cb249dbf75)

After I scroll the note down a little bit, it fits perfectly:

![image](https://github.com/chrisgurney/obsidian-note-toolbar/assets/4138869/fa1f4733-b022-4ef4-a2cc-69a2b8f63058)

Is it possible to keep its position always like in the first screenshot? In other words: The menu should not scroll when the note is scrolled up or down.

**Reply by chrisgurney** [reply]
Thank you for sharing all these screenshots!

> When I'm at the top of the note, the menu looks like this...

It looks like you're using a theme/plugin to hide the title and put it in the top heading bar like that? It _looks_ like it's a plugin, as it's showing that way for both the Minimal and Magic User themes. Can you link me to that?

> Is it possible to keep its position always like in the first screenshot? In other words: The menu should not scroll when the note is scrolled up or down.

I'm considering an enhancement that would let you set the position of each toolbar:

- below the properties/title (default as it is today)
- always stuck at the top (header)
- always at the bottom (potentially)
- as a floating action button (particularly useful for mobile)

..and you'd be able to select it by platform: desktop, mobile... and possibly by view (reading, editing).

Something you may be interested in, in the upcoming beta/release, is a command I've added that completely hides/shows the Properties section and "Properties" title. That might get you a little closer to the experience you're seeking.

**Comment by merlinuwe** [comment]
I use MagicUser Theme and I have hidden the title in obsidians settings:

![image](https://github.com/chrisgurney/obsidian-note-toolbar/assets/4138869/257f8aab-a048-480a-a3a5-7778b1236092)

The problem is the changing space between editing toolbar and yours while scrolling. (See the pink double-arrow: )

![image](https://github.com/chrisgurney/obsidian-note-toolbar/assets/4138869/19f552a9-b680-4e77-8c73-f1166d183063)

I'm looking forward to the next version with the features mentioned.

**Comment by chrisgurney** [comment]
Hi @merlinuwe 

> The problem is the changing space between editing toolbar and yours while scrolling.

I tested out the theme and it looks like there's padding outside the editable area that's outside of the Note Toolbar plugin's control, but if you really wanted to, you could adjust it with a CSS snippet.

However, you may be more interested in the [beta release](https://github.com/chrisgurney/obsidian-note-toolbar/discussions/40) which includes a `Position` setting per toolbars. Set your toolbar to `Position = Top (fixed)` and the toolbar will always be at the top of your notes. I think it fits much better with the MagicUser theme.

If you give it a go (via BRAT), let me know how it goes! If you'd prefer not to install the beta, the Community Plugins version should be updated within the week or so, depending on how testing goes.

**Comment by merlinuwe** [comment]
Here is my solution so far:

![image](https://github.com/chrisgurney/obsidian-note-toolbar/assets/4138869/61fd9637-90ca-4178-b866-8f128676dbd9)


I use the following plugin

https://github.com/c4ctus5/editor-commands-remap

to jump to the top and the bottom of the note.

For the third button, I need the following plugin

https://github.com/heycalmdown/navigate-cursor-history

to navigate to the previous caret(s).

![image](https://github.com/chrisgurney/obsidian-note-toolbar/assets/4138869/905a6d6e-6d4f-45a3-868f-66a4a70adc2e)

---

The first of these buttons jumps to the next text cursor (caret) in the history.

The following headlines are the same in some of my notes, so I can jump to them, regardless, which of these notes are open.

![image](https://github.com/chrisgurney/obsidian-note-toolbar/assets/4138869/4321ba84-32e1-4206-8703-3a9289231f72)


Hopefully the plugins used will continue to be maintained so that the configuration works permanently.

**Reply by chrisgurney** [reply]
Thank you for sharing! 

I didn't know headings worked in the URI field. Cool!
